tests/queries/0_stateless/03727_inconsistent_ast_formatting.sql

```diff
@@ -0,0 +1,3 @@
+WITH cte0d0 AS (SELECT `c0`, `c1` FROM (WITH (`t0d1`.`c1` AS `a6`) AS `a5` FROM (SELECT `c0`, `c1`, `c2` FROM (SELECT `t4d2`.`c0` AS `a3`, (6090701086911806695::Int16, `a2`, '[(-5070076742460179571,-4843538664410494325), (-66928513534.2479700937793,1591201573), (71539124302.020460009,948148932), (6604220390868324462,-1862402347)]'::Ring), minArgMin(-6203182166707188688, 'a:e:2:3:d:4:2:5'::IPv6) OVER (ROWS 0.314 FOLLOWING) AS `a4` FROM generate_series(351489, 545444) AS t0d2 FULL ANY JOIN d3.`t43` AS t1d2 ON '2163-09-25'::Date32::Date32 LEFT ARRAY JOIN (`t0d2`.`generate_series` AS `a2`) AS `a1` CROSS JOIN d1.`t20` AS t4d2)(c0, c1, c2)) AS t0d1 SELECT ngramMinHashArgCaseInsensitiveUTF8('e:e:5:f:f:a:2:d'::IPv6, `t0d1`.`c0`, `t0d1`.`c0`) AS `a7`, `t0d1`.`c0` AS `a8` GROUP BY ROLLUP(`t0d1`.`c0`) WITH TOTALS ORDER BY 1 NULLS FIRST SETTINGS short_circuit_function_evaluation = 'disable', distributed_cache_throw_on_error = 1, input_format_json_named_tuples_as_objects = 1, per_part_index_stats = 1, input_format_json_compact_allow_variable_number_of_columns = 1)(c0, c1)) SELECT CASE L1Normalize(`t0d0`.`c0`) WHEN 12850227928374013937::UInt256 THEN ['[(1109418560,-346146931325523911065012216467.6556242687905346666410885878679), (-628210000574998188,3341446455593475208), (-651458616,-899046560783798635602792564727024400945934.812427860852497035126), (-1644111332,-2203548.495352529500345770), (425653749545518952945.8848,4670716742754280848), (-2123163100,4852704288174825232.699777965443154244079455003867896122215572391124505817899493)]'::Geometry, [11726584655319463924::Int64, addSeconds(now(), -5)::DateTime::DateTime], '152.67.197.241'::IPv4, 340282366920938463463374607431768211455] END, quantileBFloat16Weighted((-(-nan))[`😆`]) OVER (ORDER BY `t0d0`.`c0` ASC, `a0` DESC ROWS UNBOUNDED PRECEDING) - toLowCardinality(([-9223372036854775808, '487d4b6f-5cd7-dd3f-da44-5241a39351c0'::UUID, 9274373979039392049::UInt64] AS `a9`)), (`t0d0`.`c0`, (`a9` AS `a10`)) FROM d2.`t27` AS t0d0 ARRAY JOIN (`t0d0`.`c0` AS `a0`) INTO OUTFILE '/var/lib/clickhouse/user_files/file.data' TRUNCATE COMPRESSION 'gz' FORMAT Null; -- { serverError NOT_IMPLEMENTED }
+SELECT (-'a')[4] FROM numbers(1) FORMAT Null; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT }
+SELECT (-('a')).1 FROM numbers(1) FORMAT Null; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT }
```

Comment:
I believe that the problem is with `(-'a').1` not with `(-('a')).1`

```suggestion
SELECT (-'a').1; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT }
```

Reply:
This one also https://pastila.nl/?00128f68/461adb1229335b074bde82da55bbcfa6#qzA28sTtB4UofoWNzbSCLw== , but I can add them all.